### ISSUE TITLE
[HELP !]When we write many kv pair to KVStore, application db will growth so fast

### ISSUE THREAD

**Issue by KamiD** [cosmos/iavl#376]
### Summary
OKExChain is a block chain which based on cosmos-sdk v39.2 and tendermint v33.9, for now, we have launched our main net and migrate ethermint into okexchain to support EVM and launched on testnet, as so far, it runs well and executed over 20 millions transactions , But during we test it , a disk growth so fast that one day growing over 5~7GB, its a bad experience for blockchain system. after some digging, we found that EVM will saving every single member of the state of contract to a kv pair and save it as iavl tree node, we have compared between okexchain and ethereum, the db size of using iavl tree node growing 10 times, the following data is ourt test case:
* deploy following contract to okexchain, call the function add() in contract via a tx
* same txs : 8000
* ethereum's db size: 167MB
* okexchain's db size: 1540MB
### Test contract:
```solidity
pragma solidity >=0.5.0 <0.8.0;

contract Storage {
    uint256 public index;
    mapping(uint256 => uint256) public map1;
    mapping(uint256 => uint256) public map2;
    mapping(uint256 => uint256) public map3;
    mapping(uint256 => uint256) public map4;
    mapping(uint256 => uint256) public map5;
    mapping(uint256 => uint256) public map6;
    mapping(uint256 => uint256) public map7;
    mapping(uint256 => uint256) public map8;
    mapping(uint256 => uint256) public map9;
    mapping(uint256 => uint256) public map10;


    function add() public {
        for (uint i = 0; i < 40; i++) {
            index += 1;
            map1[index] = index;
            map2[index] = index;
            map3[index] = index;
            map4[index] = index;
            map5[index] = index;
            map6[index] = index;
            map7[index] = index;
            map8[index] = index;
            map9[index] = index;
            map10[index] = index;
        }
    }


    function retrieve() public view returns (uint256){
        return index;
    }
}
``` 

## We deeply need help from your team to fix disk growth too fast, hope get your respond

**Comment by aaronc** [comment]
The SDK is planning to migrate away from iavl to provide better performance characteristics. Sorry you're having these problems. Wonder if you have any thoughts @robert-zaremba.

**Comment by robert-zaremba** [comment]
@KamiD  - This issue is more related to the EVM cosmos module rather than IAVL. IAVL storage is OK. That being said our work with regards to IAVL will bring significant performance improvements. 

Also, you are using v0.39 SDK. Migrating to Stargate will bring you significant improvements both in performance and storage (we  are using Protobufs rather than Amino). In 1-2 weeks we are going to make a new release (v.0.43) and I strongly recommend to use that release, because it solves few small, but important problems.

I recommend to check with the Ethermint team about the EVM module implementation. Let me know if you need a help with that.

**Comment by KamiD** [comment]
thanks for all helps, the problem was fixed by upgrading iavl version to 0.14
